src/main/java/com/rtb/projectmanagementtool/task/TaskController.java

```diff
@@ -224,6 +225,40 @@ public TaskData getTaskByID(long taskID) {
     return ancestors;
   }
 
+  public ArrayList<TaskTreeData> getTaskTree(long projectID) {
+    ArrayList<TaskTreeData> taskTree = new ArrayList<>();
+    Filter filter = new FilterPredicate("projectID", FilterOperator.EQUAL, projectID);
+    ArrayList<TaskData> tasks = getTasks(filter, NO_QUERY_LIMIT, NO_QUERY_SORT);
+    Iterator<TaskData> iterator = tasks.iterator();
+    while (iterator.hasNext()) {
```

Comment:
This part find the root?  I would suggest moving this into buildTree as well, so all tree building related logic (finding the root + adding leaves) is encapsulated in that single method